src/monexamine.cpp

```diff
@@ -129,23 +134,52 @@ bool monexamine::pet_menu( monster &z )
-        z.get_size() >= ( g->u.get_size() + 1 ) && g->u.get_weight() <= z.get_weight() / 5 ) {
-        amenu.addentry( mount, true, 'r', _( "Mount %s" ), pet_name );
-    } else if( !z.has_flag( MF_PET_MOUNTABLE ) ) {
-        amenu.addentry( mount, false, 'r', _( "%s cannot be mounted" ), pet_name );
-    } else if( z.get_size() <= g->u.get_size() ) {
-        amenu.addentry( mount, false, 'r', _( "%s is too small to carry your weight" ), pet_name );
-    } else if( g->u.get_skill_level( skill_survival ) < 1 ) {
-        amenu.addentry( mount, false, 'r', _( "You have no knowledge of riding at all" ) );
-    } else if( g->u.get_weight() >= z.get_weight() / 5 ) {
-        amenu.addentry( mount, false, 'r', _( "You are too heavy to mount %s" ), pet_name );
-    } else if( !z.has_effect( effect_saddled ) && g->u.get_skill_level( skill_survival ) < 4 ) {
-        amenu.addentry( mount, false, 'r', _( "You are not skilled enough to ride without a saddle" ) );
-    } else if( z.has_effect( effect_saddled ) && g->u.get_skill_level( skill_survival ) < 1 ) {
-        amenu.addentry( mount, false, 'r', _( "Despite the saddle, you still don't know how to ride %s" ),
-                        pet_name );
+    if( !z.has_flag( MF_RIDEABLE_MECH ) ) {
+        if( z.has_flag( MF_PET_MOUNTABLE ) && ( ( z.has_effect( effect_saddled ) &&
+                                                g->u.get_skill_level( skill_survival ) >= 1 ) || g->u.get_skill_level( skill_survival ) >= 4 ) &&
+            z.get_size() >= ( g->u.get_size() + 1 ) && g->u.get_weight() <= z.get_weight() / 5 ) {
+            amenu.addentry( mount, true, 'r', _( "Mount %s" ), pet_name );
+        } else if( !z.has_flag( MF_PET_MOUNTABLE ) ) {
+            amenu.addentry( mount, false, 'r', _( "%s cannot be mounted" ), pet_name );
+        } else if( z.get_size() <= g->u.get_size() ) {
+            amenu.addentry( mount, false, 'r', _( "%s is too small to carry your weight" ), pet_name );
+        } else if( g->u.get_skill_level( skill_survival ) < 1 ) {
+            amenu.addentry( mount, false, 'r', _( "You have no knowledge of riding at all" ) );
+        } else if( g->u.get_weight() >= z.get_weight() / 5 ) {
+            amenu.addentry( mount, false, 'r', _( "You are too heavy to mount %s" ), pet_name );
+        } else if( !z.has_effect( effect_saddled ) && g->u.get_skill_level( skill_survival ) < 4 ) {
+            amenu.addentry( mount, false, 'r', _( "You are not skilled enough to ride without a saddle" ) );
+        } else if( z.has_effect( effect_saddled ) && g->u.get_skill_level( skill_survival ) < 1 ) {
+            amenu.addentry( mount, false, 'r', _( "Despite the saddle, you still don't know how to ride %s" ),
+                            pet_name );
+        }
+    }
+    if( z.has_flag( MF_RIDEABLE_MECH ) ) {
+        itype_id bat_type = z.type->mech_battery;
+        const itype &type = *item::find_type( bat_type );
```

Comment:
```suggestion
        const itype &type = *item::find_type( z.type->mech_battery );
```
Really no need to copy that into a local variable.